tensorwaves/physics/helicityformalism/amplitude.py

```diff
@@ -501,56 +504,104 @@ def __call__(self, dataset: dict) -> tf.Tensor:
         ) * self._dynamics_function(dataset)
 
 
+class _HelicityParticle:
+    def __init__(self, name: str, helicity: int,) -> None:
+        self.name: str = name
+        self.helicity: int = helicity
+
+    @staticmethod
+    def from_dict(definition: Dict[str, Any]) -> "_HelicityParticle":
+        name = str(definition["Name"])
+        helicity = int(definition["Helicity"])
+        return _HelicityParticle(name, helicity)
+
+
+class _DecayProduct(_HelicityParticle):
+    def __init__(
+        self, name: str, helicity: int, final_state_ids: List[int],
+    ) -> None:
+        super().__init__(name, helicity)
+        self.final_state_ids: List[int] = final_state_ids
+
+    @staticmethod
+    def from_dict(definition: Dict[str, Any]) -> "_DecayProduct":
+        helicity_particle = _HelicityParticle.from_dict(definition)
+        final_state_ids = _safe_wrap_list(definition["FinalState"])
+        return _DecayProduct(
+            helicity_particle.name, helicity_particle.helicity, final_state_ids
+        )
+
+
+class _RecoilSystem:
+    def __init__(
+        self,
+        recoil_ids: Optional[List[int]] = None,
+        parent_recoil_ids: Optional[List[int]] = None,
+    ) -> None:
+        self.recoil_ids: List[int] = []
+        self.parent_recoil_ids: List[int] = []
+        if recoil_ids:
+            self.recoil_ids = recoil_ids
+        if parent_recoil_ids:
+            self.parent_recoil_ids = parent_recoil_ids
+
+    @staticmethod
+    def from_dict(definition: Dict[str, Any]) -> "_RecoilSystem":
```

Comment:
I guess those changes where necessary to get rid of the complexity issue from pylint?

Reply:
https://github.com/ComPWA/tensorwaves/pull/84#discussion_r443465116